syntax = "proto3";

package auth;

service AuthService {
    rpc Authenticate (AuthRequest) returns (AuthResponse);
    rpc ValidateToken (TokenRequest) returns (ValidationResponse);
    rpc DestroyToken (TokenRequest) returns (DestroyResponse);
}

message AuthRequest {
    string username = 1;
    string password = 2;
}

message AuthResponse {
    string token = 1;
}

message TokenRequest {
    string token = 1;
}

message ValidationResponse {
    bool valid = 1;
    string message = 2;  
    string sub = 3;
    string role = 4;
}

message DestroyResponse {
    bool success = 1;
    string message = 2;
}